File: Miscellaneous/Golang/118. Pascal's Triangle.go
package main

import "fmt"

func main() {

	result := generate(5)
	fmt.Println(result)
}

func generate(numRows int) [][]int {
	solution := make([][]int, numRows)
	i := 1

	for j := range solution {
		solution[j] = make([]int, i)
		i++
	}

	for i := 0; i < numRows; i++ {
		for j := 0; j <= i; j++ {
			if j == 0 || i == j {
				solution[i][j] = 1
			} else {
				solution[i][j] = solution[i-1][j-1] + solution[i-1][j]
			}
		}
	}

	return solution
}

File: Miscellaneous/Python/118. Pascal's Triangle.py
class Solution:
    def generate(self, numRows):
        sol = [[1]*(i+1) for i in range(numRows)]
        for i in range(2,numRows):
            for j in range(1,i):
                sol[i][j] = sol[i-1][j-1] + sol[i-1][j]

        return sol

obj = Solution()
ans = obj.generate(5)
print(ans)